### DISCUSSION TITLE
StartWithHeadlessVncPlatform  with Password

### DISCUSSION THREAD

**Discussion by Armin234** [AvaloniaUI/Avalonia#13553] [Q&A]
Hi,

is there a possibility to use StartWithHeadlessVncPlatform so that the user must enter a password ?

**Comment by maxkatz6** [comment]
Can you clarify more this question? What exact password?

**Reply by Armin234** [reply]
when the vnc client establishes a connection to the Avalonia VNC server, it would be great if the VNC server would optionally require a password for authentication, to avoid that unauthorized persons have access to the program.

**Reply by maxkatz6** [reply]
It should be possible. If you have experience with VNC, please create a PR. We have server configuration in here: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Headless/Avalonia.Headless.Vnc/HeadlessVncPlatformExtensions.cs#L37-L40

Instead of a new StartWithHeadlessVncPlatform method, we can add some boolean or enum property to the new VncPlatformOptions class.

Similarly how we have on other platforms:
Options declarations:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Windows/Avalonia.Win32/Win32PlatformOptions.cs#L96
How user can define it:
https://github.com/AvaloniaUI/Avalonia/blob/f6c5720e7a9e6a16d49dcbae32d5da8ad3433ae1/samples/RenderDemo/App.xaml.cs#L29-L32
And then in the VNC server initialization code it can be read with
`var options = AvaloniaLocator.Current.GetService<VncPlatformOptions>();`

**Reply by maxkatz6** [reply]
Although, as I see now, we already have port information in the StartWithHeadlessVncPlatform, so this concept of options is already broken there. 
Alternatively, there could be a new method:
`StartWithHeadlessVncPlatform(this AppBuilder builder, VncPlatformOptions options, string[] args = null)`

**Reply by Armin234** [reply]
Hi, thanks for your suggestion. I now wrote now my one StartWithHeadlessVncPlatform with password support.

![image](https://github.com/AvaloniaUI/Avalonia/assets/37782776/62f325d6-52c6-41b9-a05a-fc7323e9b272)
